layout/theme.liquid

```diff
@@ -12,7 +12,44 @@
       <link rel="preconnect" href="https://fonts.shopifycdn.com" crossorigin>
     {%- endunless -%}
 
-    <title>{{ shop.name }}</title>
+    {%- liquid
+      capture seo_title
+      if request.page_type == 'search' and search.performed == true
+        if search.results_count == 1
+          echo 'general.search.results_with_count.one' | t: count: search.results_count, terms: search.terms
+        elsif search.results_count > 1
+          echo 'general.search.results_with_count.other' | t: count: search.results_count, terms: search.terms
+        else
+          echo 'general.search.no_results' | t: terms: search.terms
+        endif
+      else
+        echo page_title
+      endif
+
+      if request.page_type == 'collection'
+        for filter in collection.filters
+          for value in filter.values
+            if value.active
+              echo ' / '
+              echo value.label
+            endif
+          endfor
+        endfor
+      endif
```

Comment:
Added this to see how it would work for the filtering to show up. I would still need to add something in the JS to update the page title as it updates the `updateURLHash()` if we decide to do it this way.

Reply:
Seems really granular to have theme devs update the title like this, especially as we release new features that could potentially impact it, e.g. collection filtering. I would remove for now until we have a better general sense of how we want to improve our approach to titles in our themes.